src/sleep/sleep.c

```diff
@@ -40,6 +43,161 @@
 
 static char* arg_verb = NULL;
 
+static int read_fiemap(int fd, struct fiemap **ret) {
```

Comment:
in systemd we try to make sure that functions either log about none of the errors above LOG_DEBUG level (in which case the caller is supposed to do that), or about all errors. The former is the intended scheme for "library-like" functions, and the latter for "main-like" functions, but the line is blurry when a function is clearly one or the other. 

This function logs about some errors at LOG_ERR (specifically: OOM), and about others it just does debug logging or none at all. Please clean this up: either log about all errors here, or about none/downgrade all logs to LOG_DEBUG.

Given that this kinda feels like a "library-like" API I figure it would be better to not log above LOG_DEBUG in this function.

Strictly following logging rules like this ensures that errors are logged about exactly once, as logging about them multiple times or not at all is not helpful to the user.

Reply:
OK I think I've got it how you intend now (more library like).  Please let me know if I still have something off.